javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -745,44 +745,44 @@ public static boolean isCompileTimeString(ExpressionTree node) {
 
     /** Returns the receiver tree of a field access or a method invocation. */
     public static ExpressionTree getReceiverTree(ExpressionTree expression) {
-        ExpressionTree receiver = TreeUtils.withoutParens(expression);
-
-        if (!(receiver.getKind() == Tree.Kind.METHOD_INVOCATION
-                || receiver.getKind() == Tree.Kind.MEMBER_SELECT
-                || receiver.getKind() == Tree.Kind.IDENTIFIER
-                || receiver.getKind() == Tree.Kind.ARRAY_ACCESS)) {
-            // No receiver tree for anything but these four kinds.
-            return null;
-        }
-
-        if (receiver.getKind() == Tree.Kind.METHOD_INVOCATION) {
-            // Trying to handle receiver calls to trees of the form
-            //     ((m).getArray())
-            // returns the type of 'm' in this case
-            receiver = ((MethodInvocationTree) receiver).getMethodSelect();
+        ExpressionTree receiver;
+        switch (expression.getKind()) {
+            case METHOD_INVOCATION:
+                // Trying to handle receiver calls to trees of the form
+                //     ((m).getArray())
+                // returns the type of 'm' in this case
+                receiver = ((MethodInvocationTree) expression).getMethodSelect();
 
-            if (receiver.getKind() == Tree.Kind.IDENTIFIER) {
-                // It's a method call "m(foo)" without an explicit receiver
+                if (receiver.getKind() == Tree.Kind.IDENTIFIER) {
+                    // It's a method call "m(foo)" without an explicit receiver
+                    return null;
+                } else if (receiver.getKind() == Tree.Kind.MEMBER_SELECT) {
+                    receiver = ((MemberSelectTree) receiver).getExpression();
+                } else {
+                    // Otherwise, e.g. a NEW_CLASS: nothing to do.
```

Comment:
What situation is this? How can the overall `expression` be a method invocation and the receiver be a new class tree?
Is this for a constructor invocation? I would have expected `expression` to have type `NEW_CLASS` in that case.

Reply:
I don't understand this comment either.  I think `IDENTIFIER` and `MEMBER_SELECT` are the only cases and this else branch isn't reached.  I'll make that change and see what happens.